manifests/init.pp

```diff
@@ -3,87 +3,159 @@
-  Boolean $do_ip6,
-  Optional[Integer] $edns_buffer_size,
-  Boolean $extended_statistics,
-  String $fetch_client,
-  String $group,
-  Boolean $harden_below_nxdomain,
-  Boolean $harden_dnssec_stripped,
-  Boolean $harden_glue,
-  Boolean $harden_referral_path,
-  Boolean $hide_identity,
-  Boolean $hide_version,
-  String $hints_file,
-  Optional[Integer] $infra_cache_slabs,
-  Optional[Integer] $infra_host_ttl,
-  Array $interface,
-  Boolean $interface_automatic,
-  $key_cache_size,
-  Optional[Integer] $key_cache_slabs,
-  String $keys_d,
-  Optional[String] $log_identity,
-  Boolean $log_time_ascii,
-  Boolean $log_queries,
-  Boolean $log_replies,
-  String $logdir,
-  Optional[String] $module_config,
-  Optional[String] $msg_cache_size,
-  Optional[Integer] $msg_cache_slabs,
-  Optional[Integer] $num_queries_per_thread,
-  Integer $num_threads,
-  $outgoing_interface,
-  $outgoing_port_avoid,
-  $outgoing_port_permit,
-  $outgoing_range,
-  $owner,
-  String $package_name,
-  $package_provider,
-  $port,
-  $prefetch,
-  $prefetch_key,
-  $private_domain,
-  $root_hints_url,
-  $runtime_dir,
-  $rrset_cache_size,
-  Optional[Integer] $rrset_cache_slabs,
-  $service_name,
-  Optional[String] $so_rcvbuf, #bytes, 1024, 1k, etc.
-  Optional[String] $so_sndbuf, #bytes, 1024, 1k, etc.
-  Boolean $statistics_cumulative,
-  Optional[Integer] $statistics_interval,
-  Boolean $tcp_upstream,
-  $trusted_keys_file,
-  $unwanted_reply_threshold,
-  $use_caps_for_id,
-  $val_clean_additional,
-  $val_log_level,
-  $val_permissive_mode,
-  $validate_cmd,
-  $verbosity,
-  Optional[Integer] $cache_max_ttl,
-  Optional[Integer] $cache_max_negative_ttl,
-  $custom_server_conf,
-  $skip_roothints_download,
-  Optional[String] $logfile = undef,
+  Integer[0,5]                                         $verbosity,
+  Optional[Integer]                                    $statistics_interval,
+  Boolean                                              $statistics_cumulative,
+  Boolean                                              $extended_statistics,
+  Integer[1]                                           $num_threads,
+  Integer[0, 65535]                                    $port,
+  Array[String]                                        $interface,
+  Boolean                                              $interface_automatic,
+  Optional[Array[String]]                              $outgoing_interface,           # version 1.5.10
+  Optional[Integer[1]]                                 $outgoing_range,
+  Unbound::Range                                       $outgoing_port_permit,
+  Unbound::Range                                       $outgoing_port_avoid,
+  Boolean                                              $outgoing_port_permit_first,
+  Optional[Integer[0]]                                 $outgoing_num_tcp,
+  Optional[Integer[0]]                                 $incoming_num_tcp,
+  Integer[0,4096]                                      $edns_buffer_size,
```

Comment:
i'm curious to know where this limit comes from

Reply:
in theory the ends buffer size can be 64k as per the spec however 4k is generally the limit rfc 6891 hits at this (https://tools.ietf.org/html/rfc6891#section-6.2.5) and i believe nsd, unbound,  bind, knot etc all have this value hard coded in the source code (but its been a while since i checked)

Reply:
:+1: